### PR TITLE
Add stop_timeout option to docker-module.

### PR DESCRIPTION
**Pull request by wrouesnel**
stop_timeout allows specifying a timeout for commands which can stop or restart docker containers (the time before SIGKILL is sent to the container).

The default behavior is the docker-py default (currently 10s).

I've tested and deployed a local version of this module in production, and it fills a big gap in ansible functionality with docker when working with containers with longer shutdown times.

### CODE REVIEW

**Comment by gregdek** [comment]
@cove @joshuaconner @softzilla @smashwilson Please review this PR to make sure it adheres to the following guidelines:

http://docs.ansible.com/developing_modules.html#module-checklist

If it passes these guidelines, and if you believe it’s a good PR otherwise, please add a comment with "shipit" in the text, and we will flag it for inclusion.

If it fails these guidelines, please add a comment with "needs_revision" in the text, along with the needed changes, and we will flag it for revision.

File: cloud/docker/docker.py [hunk 1]
```diff
@@ -320,6 +320,12 @@
     default: false
     aliases: []
     version_added: "2.0"
+  stop_timeout:
+    description:
+      - Set the timeout for docker stop commands before SIGKILL is sent.
```
Comment (smashwilson) [inline cloud/docker/docker.py:340]: I think it'd be friendlier to explicitly specify which actual states this is relevant for, to take out a bit of guesswork. From the source, it looks like `state: stopped` and `state: restarted`, correct?

**Comment by wrouesnel** [comment]
Rebased to include more help text about when the option applies.

**Comment by smashwilson** [comment]
:+1: Thanks! :shipit:

**Comment by gregdek** [comment]
Thanks again to @wrouesnel for this PR, and thanks @cove @joshuaconner @softzilla @smashwilson @zfil @ThomasSteinbach for reviewing. Marking for inclusion.

**Comment by gregdek** [comment]
Thanks again @wrouesnel for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts; it's possible that a merge for this functionality occurred elsewhere. If you believe this PR still applies, please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

**Comment by wrouesnel** [comment]
ready_for_review

Rebased and updated.

**Comment by smashwilson** [comment]
Great, thanks.

shipit

**Comment by gregdek** [comment]
Thanks again to @wrouesnel for this PR, and thanks @cove @joshuaconner @softzilla @smashwilson @zfil @ThomasSteinbach for reviewing. Marking for inclusion.

**Comment by gregdek** [comment]
@wrouesnel @smashwilson Apologies -- for some reason, our bot never picked up the "shipit" tag in December, and in the meantime, the code has become outdated again.

@wrouesnel if the patch still applies, please rebase and we'll get it in asap. Sorry for the confusion.

**Comment by wrouesnel** [comment]
While rebasing it looks like this patch has already been added? There is now a stop_timeout parameter (with a few changes).

**Comment by smashwilson** [comment]
Oh! Hmm. Yes, it looks like #2141 did the job. Hooray and apologies for not recognizing the duplicate.

**Comment by wrouesnel** [comment]
Closing since it was implemented by #2141.

**Pull request closed without merging**